python/dgl/data/movielens.py

```diff
@@ -0,0 +1,625 @@
+    `Movielens unplugged: experiences with an occasionally connected recommender system <https://dl.acm.org/doi/10.1145/604045.604094>`.
+
+    The datasets consist of user ratings for movies and incorporate additional user/movie information in the form of features. 
+    The nodes represent users and movies, and the edges store ratings that users assign to movies.
+
+    Statistics:
+
+    MovieLens-100K (ml-100k)
+
+    - Users: 943
+    - Movies: 1,682
+    - Ratings: 100,000 (1, 2, 3, 4, 5)
+
+    MovieLens-1M (ml-1m)
+
+    - Users: 6,040
+    - Movies: 3,706
+    - Ratings: 1,000,209 (1, 2, 3, 4, 5)
+
+    MovieLens-10M (ml-10m)
+
+    - Users: 69,878 
+    - Movies: 10,677
+    - Ratings: 10,000,054 (0.5, 1, 1.5, ..., 4.5, 5.0)
+
+    Parameters
+    ----------
+    name: str
+        Dataset name. (:obj:`"ml-100k"`, :obj:`"ml-1m"`, :obj:`"ml-10m"`). 
+    valid_ratio: int
+        Ratio of validation samples out of the whole dataset. Should be in (0.0, 1.0). 
+    test_ratio: int, optional
+        Ratio of testing samples out of the whole dataset. Should be in (0.0, 1.0) for :obj:`"ml-1m"` and :obj:`"ml-10m"`. This parameter is invalid
+        when :obj:`name` is :obj:`"ml-100k"`, since its testing samples are pre-specified. 
+        Default: None
+    raw_dir : str, optional
+        Raw file directory to download/store the data.
+        Default: ~/.dgl/
+    force_reload : bool, optional
+        Whether to re-download(if the dataset has not been downloaded) and re-process the dataset. 
+        Default: False
+    verbose : bool, optional
+        Whether to print progress information. Default: True.
+    transform : callable, optional
+        A transform that takes in a :class:`~dgl.DGLGraph` object and returns
+        a transformed version. The :class:`~dgl.DGLGraph` object will be
+        transformed before every access.
+    random_state : int, optional
+        Random seed used for random dataset split. Default: 0
+
+    Notes
+    -----
+    - When test_ratio is valid, that is, the dataset is :obj:`"ml-1m"` or :obj:`"ml-10m"`, 
+    the sum of valid_ratio and test_ratio must be in (0.0, 1.0).
+    - The number of edges is doubled to form an undirected(bidirected) graph structure.
+
+    Examples
+    --------
+    >>> from dgl.data import MovieLensDataset
+    >>> dataset = MovieLensDataset(name='ml-100k', valid_ratio=0.2)
+    >>> train_g, valid_g, test_g = dataset[0]
+    >>> train_g
+    Graph(num_nodes={'movie': 1617, 'user': 943},
+          num_edges={('movie', 'movie-user', 'user'): 64000, ('user', 'user-movie', 'movie'): 64000},
+          metagraph=[('movie', 'user', 'movie-user'), ('user', 'movie', 'user-movie')])
+
+    >>> # get ratings of edges in the training graph.
+    >>> # Note that we only store ratings in 'user-movie' edges since 'user-movie' and 'movie-user' are
+    >>> # just inverse edges and share the same set of ratings.
+    >>> ratings = train_g.edata['rate']['user-movie']
```

Comment:
```suggestion
    >>> ratings = train_g.edges["user-movie"].data["rate"]
```
The advantage of this syntax is that you can also update features by `train_g.edges["user-movie"].data["rate"] = new_ratings` but not with `train_g.edata['rate']['user-movie'] = new_ratings`.